src/webextension/manage/components/edit-item-details.js

```diff
@@ -64,9 +64,10 @@ export default class EditItemDetails extends React.Component {
 
   render() {
     const {newItem, onSave, onCancel} = this.props;
-    const controlledProps = (name) => {
+    const controlledProps = (name, maxLength = 500) => {
```

Comment:
are there tests that can be done to verify the limits are enforced?

Also, I'm not sure silently truncating the input is expected by the user.  Most other apps/sites that have limits tell you when they are exceeded.